### PR TITLE
Added dragen alignment pipeline version 4.2.4

### PR DESCRIPTION
**Pull request by alexiswl**
## TODO 

- [x] Wait for #382

### CODE REVIEW

**Comment by victorskl** [comment]
Alexis; confirming -- is this new dragen alignment pipeline is alignment QC pipeline for both WGS and WTS, correct..? And that will be replaced `wgs_alignment_qc` with Ocicat release as [discussed](https://github.com/umccr/infrastructure/pull/334#pullrequestreview-1562026653)..?

**Comment by alexiswl** [comment]
The WGS QC Pipeline has one step (which calls the dragen alignment pipeline).  

So I think we can shrink this down and just call the dragen alignment pipeline instead, we will use this for both WGS and WTS QC steps with the only difference being the '--enable-rna' parameter

**Comment by alexiswl** [comment]
Thoughts on this @skanwal, means a change to the WFL ID in https://github.com/umccr/infrastructure/pull/334

**Comment by victorskl** [comment]
[Portal Release - Ocicat: Run WTS QC workflow through Portal before running Dragen WTS Tumor Only Workflow](https://trello.com/c/H0DQyo2B/1485-portal-release-ocicat-run-wts-qc-workflow-through-portal-before-running-dragen-wts-tumor-only-workflow)

**Comment by skanwal** [comment]
I second this idea @alexiswl. Did we test dragen alignment pipeline for WTS data in dev?

**Review by skanwal — APPROVED** [review]
LGTM. Assuming it has been tested for WTS data in dev?

**Comment by alexiswl** [comment]
> Did we test dragen alignment pipeline for WTS data in dev?

Pretty sure we ran the wgs qc for WTS as a test. Can run for both WGS and WTS this afternoon

**Comment by alexiswl** [comment]
Done, with 

## WTS
wfr.82f89531a0b44d26b0e717c2f5b7178f

## WGS
wfr.a241f7e6ec06487db6a9eecc8f694470

4.2.4 didn't have the --enable-map-align-output parameter available so have added that in.  

It is a redundant parameter since we are not doing variant calling, but good to have regardless.

**Comment by alexiswl** [comment]
Also added to production_workflows project

**Comment by alexiswl** [comment]
Investigating missing RNA Quant Metrics and Transcript Coverage from multiqc in `wfr.82f89531a0b44d26b0e717c2f5b7178f` outputs. 

May be that just adding --enable-rna=True isn't enough.  

Also note that the output files for dragen still contain wgs_* prefixes :/

**Comment by alexiswl** [comment]
Ah need the --enable-rna-quantification parameter for transcript and quant metrics. This also requires an annotation file to be loaded. Are we happy to have these omitted? This 'QC' workflow can then just be with respect to alignment quality (and depth).

**Comment by skanwal** [comment]
> --enable-rna-quantification parameter for transcript and quant metrics. This also requires an annotation file to be loaded.

I'd suggest we should add these as looking at the QC report we are also missing rRNA%, which might be a useful metrics for the lab.

> Also note that the output files for dragen still contain wgs_* prefixes :/

As also mentioned yesterday - not surprised..

**Comment by alexiswl** [comment]
> I'd suggest we should add these as looking at the QC report we are also missing rRNA%, which might be a useful metrics for the lab.

We can add in `--rrna-filter-enable=true` then too - 
[from dragen v4.2 docs](https://support-docs.illumina.com/SW/dragen_v42/Content/SW/DRAGEN/rRNAfiltering_fDG_swHS.htm)

**Comment by alexiswl** [comment]
Reran with `wfr.9fca2a78578741c1b3b04fb540ff9162`.  

* %rRNA ✅ 
* Transcript Coverage Plot ✅ 
* Quant Metrics Plot ❌ 

Tried running with multiqc 1.15 (released a few days ago) but I forgot to commit the search patterns file so that's a new PR here https://github.com/ewels/MultiQC/pull/1969. 

tl;dr, we can create our own custom container if we really need the quant metrics in multiqc?

**Comment by skanwal** [comment]
> tl;dr, we can create our own custom container if we really need the quant metrics in multiqc?

For dragen-transcriptome workflow, we'll need the plot. So, it'll be good to have it added.
Thanks for the upstream PR.

**Comment by alexiswl** [comment]
This is using from this commit https://github.com/ewels/MultiQC/commit/eb98188a88cfe4d1cf718fc04669ef7d3dd7f49d

Created an image under https://quay.io/repository/umccr/multiqc?tab=tags

[PTC_NebRNA210629_dragen_alignment_multiqc (1).zip](https://github.com/umccr/cwl-ica/files/12308032/PTC_NebRNA210629_dragen_alignment_multiqc.1.zip)

**Comment by alexiswl** [comment]
Waiting for #384

**Comment by alexiswl** [comment]
@skanwal please let me know if this is all okay from your end? 

Would like to get stg, prod PRs done for WGS/WTS QC infrastructure and fix up https://github.com/umccr/data-portal-apis/pull/611 with the additional parameters discussed above.

**Comment by skanwal** [comment]
All good from my end @alexiswl.

**Review by skanwal — APPROVED** [review]
